microfrontends/app-builder-menu/src/components/Menu.tsx

```diff
@@ -54,6 +52,22 @@ export function Menu(props: Props) {
     if (config) {
       request();
     }
+    
+    const removeMenuOpenSession = () => {
+      sessionStorage.removeItem('menu_open');
+    };
+
+    if (menuOpen) {
+      window.addEventListener('beforeunload', removeMenuOpenSession);
+    }
+
+    return () => {
+      if (menuOpen) {
+        removeMenuOpenSession();
+      }
+
+      window.removeEventListener('beforeunload', removeMenuOpenSession);
```

Comment:
why the `removeEventListener` is outside the `menuOpen` if statement?

Reply:
Good catch. That should be in the `if` statement.